yarn-project/acir-simulator/src/avm/avm_memory_types.test.ts

```diff
@@ -0,0 +1,29 @@
+import { FieldValue, Uint8, Uint32 } from './avm_memory_types.js';
+
+describe('Uint8', () => {
```

Comment:
I assume more tests are coming for this?

Reply:
Yes (but not in this PR! lmk if you feel strongly). I'm thinking if it makes sense to test everything here (for the types) AND in the arithmetic+bitwise opcodes, or if there's a less verbose way. (note that there will be massive duplication).

Reply:
Sounds good i can wait until later!